nucleus/payara-modules/nucleus-microprofile/config-service/src/test/java/fish/payara/nucleus/microprofile/config/source/DirConfigSourceTest.java

```diff
@@ -77,7 +77,9 @@ public class DirConfigSourceTest {
     private static DirConfigSource source;
     private static ScheduledExecutorService exec = Executors.newScheduledThreadPool(3);
     private static ConfigProviderResolverImpl configService;
-    
+    private final int DELAY_MILLISECONDS = 10000;
```

Comment:
Delay should not be greater than 10 milliseconds. We don't want to slow down other OS, just the ones with slow file watcher. Timeout on the other hand can safely be 10 or 15 seconds.

Reply:
Delay is occurring only in case of assertion error (Line 482), which will happen mostly likely on Mac OS only

Reply:
We don't need the `while(true)` if the assertion fails once. It is safe method to try many times so the test case passes as soon as possible.